postgresql-ballerina/client.bal

```diff
@@ -0,0 +1,253 @@
+        self.clientActive = false;
+        return close(self);
+    }
+}
+
+# Provides a set of configurations for the postgresql client to be passed internally within the module.
+#
+# + host - URL of the database to connect
+# + port - Port of the database to connect
+# + user - Username for the database connection
+# + password - Password for the database connection
+# + database - Name of the database
+# + options - Postgresql datasource `Options` to be configured
+# + connectionPool - Properties for the connection pool configuration. Refer `sql:ConnectionPool` for more details
+
+type ClientConfiguration record {|
+    string host;
+    int port;
+    string? user;
+    string? password;
+    string? database;
+    Options? options;
+    sql:ConnectionPool? connectionPool;
+|};
+
+# PostgreSQL database options.
+#
+# + ssl - SSL Configuration to be used
+# + connectTimeout - The timeout value used for socket connect operations.
+#                    If connecting to the server takes longer than this value, the connection is broken. 
+#                    Value of zero means that it is disabled.
+# + socketTimeout - The timeout value used for socket read operations.
+#                   If reading from the server takes longer than this value, the connection is close
+#                   Value of zero means that it is disabled.
+# + loginTimeout - Specify how long to wait for establishment of a database connection.
+#                  Value of zero means that it is infinite.
+# + rowFetchSize - Determine the number of rows fetched in ResultSet by one fetch with trip to the database.
+# + dbMetadataCacheFields - Specifies the maximum number of fields to be cached per connection.
+#                           A value of 0 disables the cache.
+# + dbMetadataCacheFieldsMiB - Specifies the maximum size (in megabytes) of fields to be cached per connection. 
+#                            A value of 0 disables the cache.
+# + prepareThreshold - Determine the number of PreparedStatement executions required before switching over to use 
+#                            server side prepared statements.
+# + preparedStatementCacheQueries - Determine the number of queries that are cached in each connection.
+# + preparedStatementCacheSize - Determine the maximum size (in mebibytes) of the prepared queries.
+# + cancelSignalTimeout - Cancel command is sent out of band over its own connection, so cancel 
+#                         message can itself get stuck. So the timeout seconds for that.
+#                         Default value is 10 seconds
+# + tcpKeepAlive - Enable or disable TCP keep-alive probe.
+# + binaryTransfer - Use binary format for sending and receiving data if possible
+
+public type Options record {|
+  SecureSocket ssl = {};
+  decimal connectTimeout = 0;
+  decimal socketTimeout = 0;
+  decimal loginTimeout = 0;
+  int rowFetchSize?;
+  int dbMetadataCacheFields?;
+  int dbMetadataCacheFieldsMiB?;
```

Comment:
```suggestion
  int cachedMetadataFieldsCount;
  int cachedMetadataFieldSize?;
```
How about?

Reply:
I will update

Reply:
Updated. Commit id is 0fb39e1a4a35155957bdc96ffda8ad4e1dda4af0